### DISCUSSION TITLE
[SOLVED] Using TimeEdit TimeSpan and Newtonsoft.Json 12.0.3 DeserializeObject and PostAsJsonAsync

### DISCUSSION THREAD

**Discussion by gac-noob** [Megabit/Blazorise#1976] [Q&A]
Hi,

I am quite lost...

I am using Blazor Web Assembly App at the client side that consumes a ASP.NET Core Web Api Controller at the server side...
Somehow everything works fine..but when I started trying to read and write a bunch of time-related data using TimeSpan, my app crashes..sometimes I get a BadRequest (400)...???

I realized that it is related to using TimeEdit and/or TimeSpan, because if I remove all the code related to TimeEdit and TimeSpan, everything seems to work fine...

I think something is wrong somewhere when I am executing the following code:

> `var result = await http.PostAsJsonAsync("api/Employee", employee);`
> `var result = await http.PutAsJsonAsync($"api/Employee/{id}", employee);`
>` employee = JsonConvert.DeserializeObject<EmployeeVm>(await result.Content.ReadAsStringAsync());`

Can anybody please help me use TimeEdit with this? Am I using this wrong?

**Comment by stsrki** [comment]
Can you show us part of the JSON where TimeSpan is serialized?

**Reply by stsrki** [reply]
Maybe your server accepts a different date/time format.

**Reply by gac-noob** [reply]
Hi sir Mladen, you are absolutely correct...

After tinkering with my **webapi server**, I tried installing `Microsoft.AspNetCore.Mvc.NewtonsoftJson 3.1.12 `from **nuget**..
then I added `services.AddMvc().AddNewtonsoftJson();` inside the `ConfigureServices` of the **Startup.cs**..

Everything works now!

Kinda weird though...I never installed any of these in the webapi server before, and everything seems to work when I'm not using TimeSpan...hehehe

Thank you very much sir Mladen!!!

**Reply by gac-noob** [reply]
It does seem that the server is accepting a different date/time format, fortunately installing the` Microsoft.AspNetCore.Mvc.NewtonsoftJson 3.1.12` in the webapi server resolves the issue..

**Reply by stsrki** [reply]
It all depends on how the client(browser) is formating the date and time. They can match, but it's best to handle formatting explicietly yourself instead of leaving to a default format.
